### ISSUE TITLE
Empty Image control Bounds size zero

### ISSUE THREAD

**Issue by Flithor** [AvaloniaUI/Avalonia#12394]
**Describe the bug**
The Bounds of empty Image control, will return 0,0 size. but the x,y is the actual size, looks it's align to right-bottom, but the `HorizontalAlignment` and `VerticalAlignment` are `Stretch`.

**To Reproduce**
Add a auto size image control and open F12 tool to check the image control `Bounds` property.

**Expected behavior**
Bounds should have a true size.

**Screenshots**
![image](https://github.com/AvaloniaUI/Avalonia/assets/23412916/cd6b2e9c-e040-4762-9140-4bbe801ab6c6)

**Desktop (please complete the following information):**

- OS: Win 11
- Version 0.10.21

**Additional context**
will not update to 0.11.x, need current version solution.

**Comment by timunie** [comment]
I can observe the same in 11.0. However, I'm not sure if it's a bug or by design as an empty Image does not have any size by default. Maybe you can add a parent container such as a Border or Panel around it to observe the bounds of the empty image?

**Comment by Flithor** [comment]
> I can observe the same in 11.0. However, I'm not sure if it's a bug or by design as an empty Image does not have any size by default. Maybe you can add a parent container such as a Border or Panel around it to observe the bounds of the empty image?

@timunie Yes I did it in this way. But I think the control layout size should not only acording to the content size, I set aligin as `Stretch` is expected that its size will be at least adaptive to the parent. But Image control not.

**Comment by timunie** [comment]
looks like image bounds depends on `Image.Stretch`. So in that case, an Image behaves here different to other controls by design. If we would change it (if even desired), it would be a breaking change and thus cannot be accepted for either 0.10 nor 11.0. 

here is a quick sample: 

```xml
<DockPanel LastChildFill="True">

    <TextBlock DockPanel.Dock="Top">Stretch:</TextBlock>
    <ComboBox x:Name="cb_Stretch" DockPanel.Dock="Top"
              SelectedIndex="0">
        <Stretch>None</Stretch>
        <Stretch>Fill</Stretch>
        <Stretch>Uniform</Stretch>
        <Stretch>UniformToFill</Stretch>
    </ComboBox>
    
    <TextBox Watermark="Image.Bounds" UseFloatingWatermark="True" 
             DockPanel.Dock="Bottom"
             Text="{Binding #img.Bounds, Mode=OneWay}"></TextBox>
    
    <Image Source="../Assets/Sunset.jpg"
           x:Name="img"
           Stretch="{Binding #cb_Stretch.SelectedItem}"></Image>
    
</DockPanel>
```

![Animation](https://github.com/AvaloniaUI/Avalonia/assets/47110241/e9b7f172-3447-4a54-b5b2-2be4144f6f28)

**Comment by Flithor** [comment]
I know that. But this behavior is clearly inconsistent with WPF.
I will use the workaround(`Bounds` of parent `Border`) to solve it, but this is really a BUG that needs to be resolved.
